translator/translate/otel/pipeline/prometheus/translator.go

```diff
@@ -12,41 +13,107 @@ import (
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/logs/metrics_collected/prometheus"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/common"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/exporter/awsemf"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/exporter/prometheusremotewrite"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/extension/agenthealth"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/extension/sigv4auth"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/processor/batchprocessor"
+	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/processor/rollupprocessor"
 	"github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/receiver/adapter"
-)
-
-const (
-	pipelineName = "prometheus"
+	otelprom "github.com/aws/amazon-cloudwatch-agent/translator/translate/otel/receiver/prometheus"
 )
 
 type translator struct {
+	name        string
+	dataType    component.DataType
+	destination string
+}
+
+type Option func(any)
+
+func WithDestination(destination string) Option {
+	return func(a any) {
+		if t, ok := a.(*translator); ok {
+			t.destination = destination
+		}
+	}
+}
+
+func WithDataType(dataType component.DataType) Option {
+	return func(a any) {
+		if t, ok := a.(*translator); ok {
+			t.dataType = dataType
+		}
+	}
 }
 
 var _ common.Translator[*common.ComponentTranslators] = (*translator)(nil)
 
-func NewTranslator() common.Translator[*common.ComponentTranslators] {
-	return &translator{}
+func NewTranslator(opts ...Option) common.Translator[*common.ComponentTranslators] {
+	t := &translator{name: common.PipelineNamePrometheus}
+	for _, opt := range opts {
+		opt(t)
+	}
+	if t.destination != "" {
+		t.name += "/" + t.destination
+	}
+	return t
 }
 
 func (t *translator) ID() component.ID {
-	return component.NewIDWithName(component.DataTypeMetrics, pipelineName)
+	return component.NewIDWithName(t.dataType, t.name)
```

Comment:
This should always be metrics even when it's being configured in the `logs/metrics_collected` section.